app-server/app/src/main/scala/user/User.scala

```diff
@@ -105,55 +92,83 @@ object UserService {
     }
   }
 
-  def getUserById(id: java.util.UUID )(implicit database: Database): Future[Option[UsersRow]] = {
+  def createUser
+    (user: UsersRowCreate)
+    (implicit database: Database, ec: ExecutionContext): Future[UsersRow] =
+  {
     import database.driver.api._
+    val userRow = user.toUsersRow()
 
     database.db.run {
-      Users.filter(_.id === id).result.headOption
+      Users.forceInsert(userRow)
+    } map {
+      x => userRow
     }
   }
 
-  def createUser(user: UsersRowCreate)
-    (implicit database: Database, ec: ExecutionContext): Future[UsersRow] = {
+  def createUserFromJwt
+    (token: JsonWebToken)
+    (implicit database: Database, ec: ExecutionContext): Future[Option[UsersRow]] =
+  {
     import database.driver.api._
-    val userRow = user.toUsersRow()
 
+    val newUUID = java.util.UUID.randomUUID
     database.db.run {
-      Users.forceInsert(userRow)
-    } map {
-      x => userRow
+      Organizations.filter(_.name === "Public").result.headOption
+    } flatMap {
+      org => org match {
+        case Some(org) => {
```

Comment:
I think it would be good to break up some of this deeply nested structuring by setting some `val`s and other techniques. It's kind of hard to follow.